app/controllers/events_controller.rb

```diff
@@ -93,6 +100,80 @@ def schedule
     @schedule_type = @event.type.downcase!
   end
 
+  # GET /events/1/ranking
+  def ranking
+    # Array of RankingEntry Structs that gets sorted when filled completely
+    @ranking_entries = []
+
+    # TODO Either add the possibility to calculate rankings for single-player events
+    # TODO or replace `teams` with the to be introduced EventParticipant superclass
+    # Leaves the Array of RankingEntry Structs empty when no teams participate in the event
+    @event.teams.each do |team|
+      ranking_entry = RankingEntry.new(nil, team.name, 0, 0, 0, 0, 0, 0, 0, 0)
+
+      event_matches = @event.matches
+      # Considers only the team's home matches that belong to the event
+      home_matches_in_event = team.home_matches & event_matches
+      parse_matches_data_into_ranking_entry team, ranking_entry, home_matches_in_event, :parse_match_details_for_home
+
+      # Considers only the team's away matches that belong to the event
+      away_matches_in_event = team.away_matches & event_matches
+      parse_matches_data_into_ranking_entry team, ranking_entry, away_matches_in_event, :parse_match_details_for_away
+
+      ranking_entry.goals_difference = ranking_entry.goals - ranking_entry.goals_against
+      @ranking_entries.push ranking_entry
+    end
+
+    # Sorts the RankingEntries in the following order:
+    #   1. DESCENDING by points
+    #   2. DESCENDING by goals
+    #   3. ASCENDING by name
+    @ranking_entries = @ranking_entries.sort_by { | ranking_entry | [-ranking_entry.points, -ranking_entry.goals, ranking_entry.name] }
+
+    # Adds a rank to each RankingEntry based on its position in the Array
+    @ranking_entries.each_with_index do |ranking_entry, index|
+      ranking_entry.rank = index + 1
+    end
+  end
+
+  def parse_matches_data_into_ranking_entry(team, ranking_entry, matches, parse_match_details_for_home_or_away)
+    matches.each do |match|
+      score_home = match.score_home
+      score_away = match.score_away
+      match_has_score = score_home && score_away
+      next unless match_has_score
+
+      ranking_entry.match_count += 1
+      parse_match_result_into_ranking_entry team, match, ranking_entry
+
+      send(parse_match_details_for_home_or_away, ranking_entry, score_home, score_away, match)
+    end
+  end
+
+  def parse_match_result_into_ranking_entry(team, match, ranking_entry)
```

Comment:
All these methods don't really belong to the controller. Try to keep as much logic as possible outside of the controller. Methods that aren't actions need a good reason to even exist. And in this case it is very model specific. I would suggest encapsulation these methods into the Rankinglist model or even better into seperate plain old ruby objects (POROs) that handle all the neseccary logic.
The ranking function should do much more than setting `@ranking_entries = @event.ranking_entries`.

Reply:
Here is a somewhat related article on the matter of POROs: http://blog.steveklabnik.com/posts/2011-09-06-the-secret-to-rails-oo-design

Reply:
Thanks! Refactoring will follow up:  https://github.com/hpi-swt2/sport-portal/issues/367